### PR TITLE
Use Qt 6.5 styleHint colorScheme to detect dark theme

### PR DESCRIPTION
**Pull request by metsma**
- keep linux compatibility

WE2-826

Signed-off-by: Raul Metsma <raul@metsma.ee>

### CODE REVIEW

**Review by mrts — CHANGES_REQUESTED** [review]

File: src/controller/application.cpp [hunk 1]
```diff
@@ -70,23 +71,23 @@ Application::Application(int& argc, char** argv, const QString& name) :
...
 bool Application::isDarkTheme()
 {
-#ifdef Q_OS_WIN
-    QSettings settings(
-        "HKEY_CURRENT_USER\\Software\\Microsoft\\Windows\\CurrentVersion\\Themes\\Personalize",
-        QSettings::NativeFormat);
-    return settings.value("AppsUseLightTheme", 1).toInt() == 0;
+#if QT_VERSION >= QT_VERSION_CHECK(6, 5, 0)
+    return styleHints()->colorScheme() == Qt::ColorScheme::Dark;
 #else
```
[the remark this answers is not in the document]
Comment (metsma) [reply src/controller/application.cpp:84]: Why exclude other OS-es where gsettings IS available?

[the remark this answers is not in the document]
Comment (mrts) [reply src/controller/application.cpp:82]: Are there any OSes like this? I would argue that we should suggest them to use Qt v6.5+. Defaulting to `gsettings` as a general, OS-independent fallback looks fragile to me.

[the remark this answers is not in the document]
Comment (metsma) [reply src/controller/application.cpp:82]: BSD-s and so on?

[the remark this answers is not in the document]
Comment (mrts) [reply src/controller/application.cpp:82]: Good point, perhaps we should use `#elif defined(Q_OS_UNIX)` then which is defined on both Linux and BSDs as a defensive measure?

**Review by mrts — APPROVED** [review]

**Pull request merged**